lib/services/rpcbind.pm

```diff
@@ -57,12 +57,20 @@ sub check_enabled {
 sub check_service {
     common_service_action('rpcbind.socket', $service_type, 'is-enabled');
     common_service_action('rpcbind.socket', $service_type, 'is-active');
+    common_service_action($nfs_server, $service_type, 'is-enabled');
+    common_service_action($nfs_server, $service_type, 'is-active');
 }
 
 sub check_function {
     assert_script_run("rpcinfo");
     # Wait for updated rpcinfo.
-    sleep(5);
+    if (script_run('rpcinfo | grep nfs') != 0) {
+        record_soft_failure 'bsc#1193028 - rpcinfo can not get nfs info in time on s390x'
+          for (1 .. 10) {
+            last unless script_run('rpcinfo | grep nfs') != 0;
```

Comment:
Could you use positive condition for the judgement?
Such as :
last if script_run('rpcinfo | grep nfs') == 0;

Too many negative conditions make others hard to understand the code.

Reply:
changed

Reply:
changed

Reply:
changed